vmc/resource_vmc_cluster_test.go

```diff
@@ -172,3 +173,22 @@ func testAccVmcClusterResourceImportStateIdFunc(resourceName string) resource.Im
 		return fmt.Sprintf("%s,%s", rs.Primary.ID, rs.Primary.Attributes["sddc_id"]), nil
 	}
 }
+
+func TestBuildClusterConfig(t *testing.T) {
+	testBareBonesClusterConfig(t)
+}
+
+func testBareBonesClusterConfig(t *testing.T) {
+	var mockRawData = map[string]interface{}{
+		"num_host":           MinHosts,
+		"host_instance_type": HostInstancetypeI4I,
+	}
+	var mockResourceSchema = schema.TestResourceDataRaw(t, clusterSchema(), mockRawData)
+
+	bareBonesClusterConfig, _ := buildClusterConfig(mockResourceSchema)
+
+	if *bareBonesClusterConfig.HostInstanceType != model.SddcConfig_HOST_INSTANCE_TYPE_I4I_METAL {
```

Comment:
Can we have a table test to test all instance types, not just HostInstancetypeI4I?

+ We might also want to introduce some test assertions instead (here and other tests)? (easier to read I think -  like https://pkg.go.dev/github.com/stretchr/testify/assert)

Reply:
Great many thanks! I've been looking for TestNG/JUnit's assertEquals  equivalent for GO, but some unit tests at the aws terraform provider were using this lame equality test.